applications/asset_tracker_v2/src/cloud/cloud_codec/nrf_cloud_codec.c

```diff
@@ -104,14 +112,103 @@ static int add_data(cJSON *parent, const char *app_id, const char *str_val,
 		goto exit;
 	}
 
-	json_add_obj(parent, object_name, data_obj);
+	if (object_name == NULL) {
+		json_add_obj_array(parent, data_obj);
+	} else {
+		json_add_obj(parent, object_name, data_obj);
+	}
+
 	return 0;
 
 exit:
 	cJSON_Delete(data_obj);
 	return err;
 }
 
+static int add_batch_data(cJSON *array, enum batch_data_type type, void *buf, size_t buf_count)
+{
+	for (int i = 0; i < buf_count; i++) {
+		switch (type) {
+		case GPS: {
+			int err;
+			struct cloud_data_gps *data = (struct cloud_data_gps *)buf;
+
+			err =  add_data(array, APP_ID_GPS, data[i].nmea,
+					&data[i].gps_ts, data[i].queued, NULL);
+			if (err && err != -ENODATA) {
+				return err;
+			}
+
+			data[i].queued = false;
+			break;
+		}
+		case ENVIRONMENTALS: {
+			int err;
+			char humidity[10];
+			char temperature[10];
+			struct cloud_data_sensors *data = (struct cloud_data_sensors *)buf;
+			int64_t ts_temp = data[i].env_ts;
+
+			snprintk(humidity, sizeof(humidity), "%f", data[i].hum);
```

Comment:
I think we can specify the number of decimals we want here (and elsewhere). We will probably not overflow the buffer anyway, but nice to make sure of it. For instance `%.2f`. Return values could also be checked to ensure that we input a null-terminated buffer to `add_data`.

Reply:
Good point. `%.2f` should be sufficient for all use cases in the app.

Reply:
Ill update the PR and add a separate commit that addresses this.